### ISSUE TITLE
Adding reactions to activities on many users feeds

### ISSUE THREAD

**Issue by Crysknight** [GetStream/stream-js#400]
Our feed architecture looks like that: we have different feed types (e. g. user's feed, project's feed, event's feed). Users can subscribe to any of those feeds which makes their timeline feed follow the target feed. This renders the user's timeline feed as the summary feed and the main interaction point with the feed system.

## What problem does this feature solve
When adding reactions (likes, comments) to an activity on the timeline feed, we are basically adding a reaction to a copied activity with a different id than the original. On other feeds including other users' timeline feeds and the feed where the original activity comes from, those reactions do not appear. This makes the user's timeline feed a personal little pool with comments and likes that nobody actually sees which is obviously not the expected behavior.

## Solutions
The documentation suggests using `targetFeeds` option when adding reactions. This poses a problem where I have to identify all the feeds that have a particular activity on them. I have to move my reactions logic on the backend (since the lib on the frontend has lowered permissions which is totally fine) then, with each reaction I have to request followers of the original feed in a paginated fashion with a limit of 1000 followers.
A propagation by `foreign_id` would be much simpler and reliable. By specifying the `foreign_id`, reactions could be copied over to the activities on the other feeds. Or else when requesting reactions' list, `foreign_id` could be specified returning the corresponding reactions (that also would require some way to correctly request reactions and reaction counts in `feed.get` method).

Maybe I could do something with the architecture of the feed or maybe I'm doing something wrong with `targetFeeds` option? Please, point it out. I've read the documentation regarding reactions but it doesn't seem to work with our architecture.

gz#7679

**Comment by shodgetts** [comment]
Agent comment from Stephen Hodgetts in <a href='https://getstream.zendesk.com/agent/tickets/7679'>Zendesk ticket #7679</a>:<br><br>Hello, whats your appID or key please? Looking into getting you some help shortly.

**Comment by ferhatelmas** [comment]
@Crysknight important question is that 

> we are basically adding a reaction to a copied activity with a different id than the original

why do you need to copy activities? Architecture of feeds sounds off. Activities normally go to main feeds and timelines follow these main feeds (events, users, etc.) then activity id on timeline will be same with the activity id in the original feed such that any reaction will be on both.

**Comment by Crysknight** [comment]
@ferhatelmas @shodgetts 
Double-checked that activity copying thing. That was a problem on our side all along. Because I was posting the same activity into different feeds filtering duplicates in the timeline feed by foreign_id, I ended up with activities of the same foreign_id but different activityIds. And mistakenly assumed that followers feeds receive a copy of the activity. I had to double-check that first, sorry.
